### PR TITLE
#65 - Add MinIO (Backend)

### LINKED ISSUE #65: Add MinIO (Backend)

### PR DESCRIPTION
**Pull request by NN171**
In PR I have added MinIO to store images of profiles and directions. Also changed types of caching in directions and profiles.
Added global CORS config

### CODE REVIEW

File: backend/BackendMIIT/src/main/resources/application.yml [hunk 1]
```diff
@@ -26,6 +26,15 @@ spring:
...
 
+minio:
+  bucket: profile-images
+  url: http://minio:9000
+  access:
+    key: minioadmin
+  secret:
+    key: minioadmin
+
+
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/resources/application.yml]: extra empty line

File: backend/BackendMIIT/docker-compose.yml [hunk 2]
```diff
@@ -41,3 +45,16 @@ services:
...
+    container_name: minio
+    environment:
+      - MINIO_ROOT_USER=minioadmin
+      - MINIO_ROOT_PASSWORD=minioadmin
+    command: server /minio --console-address :9001
+    ports:
+      - '9000:9000'
+      - '9001:9001'
+    volumes:
+      - minio-data:/minio
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/docker-compose.yml]: Add the empty line in the end

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/configuration/WebConfig.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@EnableWebMvc
+@Configuration
+public class WebConfig implements WebMvcConfigurer {
+
+	@Override
+	public void addCorsMappings(CorsRegistry registry) {
+		registry
+				.addMapping("/api/miit")
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/configuration/WebConfig.java]: Remove extra tabs

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java:1]: Reformat the order of the methods:
1) bucketExists
2) createBucket
3) generateFileName 

etc.
This structure helps the reader to see sequence of methods used.

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		return imageUrl;
+	}
+
+	private String generateFileName(MultipartFile file) {
+		String extension = getExtension(file);
+		return UUID.randomUUID() + "." + extension;
+	}
+
+	private String getExtension(MultipartFile file) {
+		return file.getOriginalFilename().
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java]: Move "." to the next line:
```
return file.getOriginalFilename()
    .substring(file.getOriginalFilename().lastIndexOf(".") + 1);
```

**Review by ImmortalIdiot — COMMENTED** [review]
Fix the tabs

**Comment by ImmortalIdiot** [comment]
News:
image

Upload results with ppType - "min", category - "Общий конкурс", page - 4: [Sort direction with paggination.txt](https://github.com/user-attachments/files/17706813/Sort.direction.with.paggination.txt)

File: backend/BackendMIIT/src/main/resources/application.yml (hunk 1 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/resources/application.yml]: Added

File: backend/BackendMIIT/docker-compose.yml (hunk 2 above)
Comment (NN171) [reply backend/BackendMIIT/docker-compose.yml]: Added

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/configuration/WebConfig.java (hunk 3 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/configuration/WebConfig.java]: Removed

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java:1]: Reordered

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java (hunk 4 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/MinioServiceImpl.java]: Added

**Review by ImmortalIdiot — APPROVED** [review]

**Pull request merged**